src/content_scripts/autocorrect.js

```diff
@@ -0,0 +1,400 @@
+
+    if(isSuccess) {
+        return;
+    }
+
+    // Firefox input and textarea fields: https://bugzilla.mozilla.org/show_bug.cgi?id=1220696
+    if (typeof target.setRangeText === "function") {
+        const start = target.selectionStart;
+        const end = target.selectionEnd;
+
+        if (start !== undefined && end !== undefined) {
+            target.setRangeText(atext);
+
+            target.selectionStart = target.selectionEnd = start + atext.length;
+
+            // Notify any possible listeners of the change
+            const event = document.createEvent("UIEvent");
+            event.initEvent("input", true, false);
+            target.dispatchEvent(event);
+
+            return;
+        }
+    }
+
+    throw new Error("nothing selected");
+}
+
+/**
+ * Insert into page.
+ *
+ * @param {string} atext
+ * @returns {void}
+ */
+function insertIntoPage(atext) {
+    return insertCaret(document.activeElement, atext);
+}
+
+/**
+ * Count Unicode characters.
+ * Adapted from: https://blog.jonnew.com/posts/poo-dot-length-equals-two
+ *
+ * @param {string} str
+ * @returns {number}
+ */
+function countChars(str) {
+    // removing the joiners
+    const split = str.split("\u{200D}");
+    let count = 0;
+
+    for (const s of split) {
+        // removing the variation selectors
+        count += Array.from(s.split(/[\ufe00-\ufe0f]/).join("")).length;
+    }
+
+    return count;
+}
+
+/**
+ * Delete at caret.
+ *
+ * @param {Object} target
+ * @param {string} atext
+ * @returns {void}
+ */
+function deleteCaret(target, atext) {
+    const count = countChars(atext);
+    if (count > 0) {
+        const isSuccess = document.execCommand("delete", false);
+        if (isSuccess) {
+            for (let i = 0; i < count - 1; ++i) {
+                document.execCommand("delete", false);
+            }
+        }
+        // Firefox input and textarea fields: https://bugzilla.mozilla.org/show_bug.cgi?id=1220696
+        else if (typeof target.setRangeText === "function") {
+            const start = target.selectionStart;
+
+            target.selectionStart = start - atext.length;
+            target.selectionEnd = start;
+            target.setRangeText("");
+
+            // Notify any possible listeners of the change
+            const e = document.createEvent("UIEvent");
+            e.initEvent("input", true, false);
+            target.dispatchEvent(e);
+        }
+    }
+}
+
+/**
+ * Convert fractions and constants to Unicode characters.
+ * Adapted from: https://github.com/tdulcet/Tables-and-Graphs/blob/master/graphs.hpp
+ *
+ * @param {number} anumber
+ * @param {number} afraction
+ * @returns {string}
+ */
+function outputLabel(anumber, afraction) {
+    let output = false;
+
+    const number = parseFloat(anumber);
+    let intpart = Math.trunc(number);
+    const fractionpart = afraction ? parseFloat(afraction) : Math.abs(number % 1);
+
+    let strm = "";
```

Comment:
Sorry, what? Please use self-explaining variable names.
Only use abbrevations when they are obvious/common.

Reply:
Oh, it was short for stream, since I ported this function from C++ and the variable was of the `stringstream` type. Since in JS it is now a string, I will change the name to just `str`.